webapp/src/compiler.ts

```diff
@@ -341,7 +343,22 @@ export function pySnippetArrayToBlocksAsync(code: string[], blockInfo?: ts.pxtc.
             let ts = "";
             for (let file of files) {
                 let match = res.outfiles[file + ".ts"].match(namespaceRegex);
-                if (match && match[1]) ts += `{\n${match[1]}\n}\n`
+                if (match && match[1]) {
+                    let noNamespace = match[1];
+                    let lines = noNamespace.split("\n");
+                    let cleanLines = lines.map((element: string)=>{
+                        match = element.match(exportLetRegex);
+                        let strippedExportLet;
+                        match? strippedExportLet = match[1] : strippedExportLet = element;
+                        match = strippedExportLet.match(exportFunctionRegex);
```

Comment:
maybe worth creating a new const for each math? It ends up feeling a bit hard to track for me when we reuse match variables but that's just me~

Reply:
yeah I'll do that! I did it both ways, but switched it to this since I wasn't sure what would be preferred :)